feat: Home Page

sunputawan opened the pull request:
## Issue ID/Name
<!--Linear card ID-->
https://linear.app/oddsnends/issue/ODD-12/[fe]-home-page

## Description
<!--What does this PR do? What is being changed, added, or removed? Does it introduce any breaking changes?-->
- Built and Styled Home Page (without sliding box yet)
- Created SponserSection.tsx within /src/app/components/SponserSection and placed it inside the SocialMediaBar.tsx
- Defined custom colors in `tailwind.config.ts` file to be reused for entire project
- Added `flex-grow` as a className for `children` in the Root Layout to ensure that the `SocialMediaBar` stays at the bottom of the entire screen, even when the page content is not enough to fill the viewport.
- For ERROR: GET 400 because there is no src for images yet

## Checklist
<!--Put the 'X' inside the [] to check. Most of the time, everything should be checked.-->
- [X] I have self-reviewed my code.
- [X] I have commented on my code, particularly in the hard-to-understand area.
- [ ] (Should Be) I have NOT committed the modified environment/config file(s) to this PR.
- [X] I have tagged a reviewer for this PR.

## Additional configuration/environment
in tailwind.config.ts, added
'custom-dark-gray': '#353535'
'custom-light-gray': '#C4C4C4'
referenced from our project's Figma

## Screenshot(s)
<!--Add your screenshot(s) here, caption for the image is recommended.-->


### The Home Page
<img width="388" alt="Screenshot 2568-01-31 at 06 27 01" src="https://github.com/user-attachments/assets/2110393e-772a-4ae7-978f-fa594ec088ce" />

### Custom Colors Defined
<img width="484" alt="Screenshot 2568-02-01 at 03 51 15" src="https://github.com/user-attachments/assets/8b1dfb74-1eea-4ce1-9f9d-03cbe22eab17" />

### `flex-grow` Added to `children` in Root Layout
<img width="704" alt="Screenshot 2568-02-01 at 03 52 34" src="https://github.com/user-attachments/assets/034b0e95-3a60-4edc-ae8a-4777053a4078" />

### WITHOUT `flex-grow`
<img width="345" alt="Screenshot 2568-02-01 at 04 02 54" src="https://github.com/user-attachments/assets/28e313d7-2313-4815-8c7c-728acc1ef66a" />

### WITH `flex-grow`
<img width="335" alt="Screenshot 2568-02-01 at 03 58 12" src="https://github.com/user-attachments/assets/a059aaa7-4cb5-435e-b705-040f7ec9ddd1" />


## Reviewer
<!--Person who needs to review your PR-->
@LittleNa1000 

## Remark
<!--Further information about this PR-->
Added `flex-grow` to the `children` in RootLayout instead of Added `h-screen` to the HomePage because adding `h-screen` to the HomePage would result in the total height of `100vh + SocialMediaBar's height` , as SocialMediaBar is placed inside the RootLayout

LittleNa1000 commented:
Deploy preview for _oddsnends_ ready!

✅ Preview
https://oddsnends-5slxrbn54-oddsnends-projects.vercel.app

Built with commit aa0b5a435c6417ac3bb9aafaf4329ebc4d4c356d.
This pull request is being automatically deployed with [vercel-action](https://github.com/marketplace/actions/vercel-action)

LittleNa1000 commented on src/app/home/page.tsx:
```diff
@@ -0,0 +1,55 @@
+
+export default function HomePage() {
+  return (
+    <div className="">
```
why this div is needed?

LittleNa1000 commented on src/app/components/SponserSection/SponsorSection.tsx:
```diff
@@ -0,0 +1,35 @@
+
+const SponsorSection = () => {
+  return (
+    <section className="mt-16 p-6 bg-white">
```
not sure if `mt-16` is taking too much space kub.
maybe remove `mt-16` and add `mb-16` in `home/page.tsx` instead

sunputawan commented:
@LittleNa1000 
I've fixed the request changes, including removing `mt-16` in SponserSection and instead placing `mb-16` at the HomePage, also removed the unnecessary `div`

LittleNa1000 reviewed (approved):
good job. very detailed pr description kub.
